### ISSUE TITLE
magick xc:white white.pnm creates a black image

### ISSUE THREAD

**Issue by cjmayo** [ImageMagick/ImageMagick#1817]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Previous similar issues: #1632, #1636 

### Steps to Reproduce

```
magick -size 100x100 xc:white white.pnm
magick display white.pnm
```
### System Configuration

- ImageMagick version: 7.0.9-8, 7.0.9-10 and 7.0.9-12
- Environment (Operating system, version and so on): Gentoo Linux
- Additional information:

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by cjmayo** [comment]
Commit applied as a patch to 7.0.9-12 did fix the problem and 7.0.9-13 was OK.

Problem is back in 7.0.9-14 and also `magick -size 100x100 xc:black black.pnm` creates a white image.

**Comment by fmw42** [comment]
I can confirm that black.pnm as created above in 7.0.9.14 does display as white in ImageMagick magick display and also in GIMP.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by cjmayo** [comment]
7.0.9-15 has the issue for black and white.

**Comment by urban-warrior** [comment]
Fixed in 7.0.9-16.  Thanks.

**Comment by cjmayo** [comment]
Yes indeed, 7.0.9-16 is working. Thanks!
